From 1ea2a8ef86b98be40c77bf6a95d2bc09e67e4884 Mon Sep 17 00:00:00 2001
From: Alessandro Decina <alessandro.decina@collabora.co.uk>
Date: Thu, 22 Sep 2011 18:47:24 +0200
Subject: [PATCH 087/119] pvrvideosink: restrict input to yuv-strided for now

-rgb and non strided -yuv somewhat work but are broken for some resolutions and
have awful performances.
---
 sys/pvr2d/gstpvrvideosink.c |   11 ++++++++---
 1 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/sys/pvr2d/gstpvrvideosink.c b/sys/pvr2d/gstpvrvideosink.c
index 2c67c9c..4c39638 100644
--- a/sys/pvr2d/gstpvrvideosink.c
+++ b/sys/pvr2d/gstpvrvideosink.c
@@ -70,9 +70,14 @@ static GstStaticPadTemplate gst_pvrvideosink_sink_template_factory =
     GST_STATIC_PAD_TEMPLATE ("sink",
     GST_PAD_SINK,
     GST_PAD_ALWAYS,
-    GST_STATIC_CAPS (GST_VIDEO_CAPS_YUV_STRIDED ("NV12", "4096") ";"
-        GST_VIDEO_CAPS_BGRx)
-    );
+    GST_STATIC_CAPS (
+        "video/x-raw-yuv-strided, "                \
+        "format = (fourcc) NV12, "                 \
+        "rowstride = (int) 4096, "                 \
+        "width = " GST_VIDEO_SIZE_RANGE ", "       \
+        "height = " GST_VIDEO_SIZE_RANGE ", "      \
+        "framerate = " GST_VIDEO_FPS_RANGE
+    ));
 
 enum
 {
-- 
1.7.5.4

